File: temp/121.best_time_to_buy_and_sell_stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        if n == 0 or n == 1:
            return 0
        min_price = prices[0]
        max_profit = 0
        for i in range(1, n):
            max_profit = max(max_profit, prices[i] - min_price)
            min_price = min(min_price, prices[i])
        return max_profit
        

File: C++/Dynamic Programming/121. Best Time to Buy and Sell Stock.cpp
// 系列问题 121 122 123 188 309 714

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int currMin = prices[0];
        for (int i = 1; i < prices.size(); i++) {
            res = max(res, prices[i] - currMin);
            currMin = min(currMin, prices[i]);
        }
        return res;
    }
};